packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,221 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import {
+  ASSET_NAMESPACE,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  fromAssetId,
+  solAssetId,
+  toAssetId,
+  wrappedSolAssetId,
+} from '@shapeshiftoss/caip'
+import type { GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { bnOrZero, convertDecimalPercentageToBasisPoints } from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { v4 as uuid } from 'uuid'
+
+import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
+import type {
+  CommonTradeQuoteInput,
+  GetTradeRateInput,
+  ProtocolFee,
+  SwapErrorRight,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../../types'
+import { SwapperName, TradeQuoteError } from '../../../types'
+import { getRate, makeSwapErrorRight } from '../../../utils'
+import { getJupiterQuote, isSupportedChainId } from '../utils/helpers'
+
+const _getTradeQuote = async (
+  input: CommonTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps,
+    receiveAddress,
+    accountNumber,
+    slippageTolerancePercentageDecimal,
+  } = input
+
+  const { assetsById } = deps
+
+  const jupiterUrl = deps.config.REACT_APP_JUPITER_API_URL
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  const maybeQuoteResponse = await getJupiterQuote({
+    apiUrl: jupiterUrl,
+    sourceAsset: sellAsset.assetId === solAssetId ? wrappedSolAssetId : sellAsset.assetId,
+    destinationAsset: buyAsset.assetId === solAssetId ? wrappedSolAssetId : buyAsset.assetId,
```

Comment:
Jupiter use wrapped SOL as an asset to swap with SOL, it auto unwrap/wrap in the instructions, this is how it works!

Reply:
Very nice catch, It's supported but I need to disable the auto wrap feature in that case, updated in [4e76e81](https://github.com/shapeshift/web/pull/8120/commits/4e76e8115223d91bd602f7d9ceed80c94c77ea15) , if the buyAsset is wrapped sol, don't unwrap automatically

Reply:
But there are magic somewhere that prevent from selecting wapped sol for some reasons

Reply:
LGTM! Will take a look at selection when testing this at runtime